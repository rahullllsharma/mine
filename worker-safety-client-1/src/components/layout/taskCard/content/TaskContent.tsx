import type { Hazard } from "@/types/project/Hazard";
import type { Control } from "@/types/project/Control";
import HazardCard from "../hazard/HazardCard";
import ControlCard from "../control/ControlCard";

function ControlCards({ controls }: { controls: Control[] }) {
  return (
    <>
      {controls.map(({ id, name }) => (
        <ControlCard className="mt-2 first:mt-0" key={id} label={name} />
      ))}
    </>
  );
}

type HazardCardProps = {
  hazards: Hazard[];
};

export default function TaskContent({
  hazards = [],
}: HazardCardProps): JSX.Element {
  return (
    <div className="bg-brand-gray-10 px-4 py-2">
      {hazards.map(({ id, name, controls }) => (
        <HazardCard key={id} header={name}>
          <ControlCards controls={controls} />
        </HazardCard>
      ))}
    </div>
  );
}
