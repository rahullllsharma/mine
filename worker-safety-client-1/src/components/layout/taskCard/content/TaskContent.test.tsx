import { render, screen } from "@testing-library/react";
import TaskContent from "./TaskContent";

describe("TaskContent", () => {
  describe("when task don't contain hazards", () => {
    it("should not render a hazard card", () => {
      render(<TaskContent hazards={[]} />);
      const hazardCardElements = screen.queryAllByTestId("hazardCard");
      expect(hazardCardElements).toHaveLength(0);
    });
  });

  describe("when task contain hazards", () => {
    const hazards = [
      {
        id: "1",
        name: "hazard 123",
        isApplicable: true,
        controls: [
          {
            id: "1",
            name: "control name",
            isApplicable: true,
          },
          {
            id: "2",
            name: "control name 2",
            isApplicable: true,
          },
        ],
      },
      {
        id: "2",
        name: "hazard 123",
        isApplicable: true,
        controls: [
          {
            id: "1",
            name: "control name",
            isApplicable: true,
          },
          {
            id: "2",
            name: "control name 2",
            isApplicable: true,
          },
        ],
      },
    ];

    it("should render a list of hazards and controls", () => {
      const { asFragment } = render(<TaskContent hazards={hazards} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it(`should render ${hazards.length} total hazard card(s)`, () => {
      render(<TaskContent hazards={hazards} />);
      const hazardCardElements = screen.getAllByTestId("hazardCard");
      expect(hazardCardElements).toHaveLength(hazards.length);
    });

    describe("when there is no controls", () => {
      it("should not render a control card", () => {
        const hazardsWithoutControls = [
          {
            id: "1",
            name: "hazard 123",
            isApplicable: true,
            controls: [],
          },
        ];
        render(<TaskContent hazards={hazardsWithoutControls} />);
        const controlCardElements = screen.queryAllByTestId("controlCard");
        expect(controlCardElements).toHaveLength(0);
      });
    });

    describe("when we have a set of controls", () => {
      it(`should render ${hazards[0].controls.length} control cards`, () => {
        const hazard = hazards[0];
        render(<TaskContent hazards={[hazard]} />);
        const controlCardElements = screen.queryAllByTestId("controlCard");
        expect(controlCardElements).toHaveLength(hazard.controls.length);
      });
    });
  });
});
