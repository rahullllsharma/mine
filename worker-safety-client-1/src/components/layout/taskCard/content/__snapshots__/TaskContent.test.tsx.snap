// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContent when task contain hazards should render a list of hazards and controls 1`] = `
<DocumentFragment>
  <div
    class="bg-brand-gray-10 px-4 py-2"
  >
    <div
      class="text-base text-neutral-shade-100 mb-2"
      data-testid="hazardCard"
    >
      <header
        class="flex flex-1 text-brand-gray-80 py-2 pr-2 mr-1 font-semibold"
      >
        hazard 123
      </header>
      <div
        class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded mt-2 first:mt-0"
        data-testid="controlCard"
      >
        <span
          class="flex-1"
        >
          control name
        </span>
      </div>
      <div
        class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded mt-2 first:mt-0"
        data-testid="controlCard"
      >
        <span
          class="flex-1"
        >
          control name 2
        </span>
      </div>
    </div>
    <div
      class="text-base text-neutral-shade-100 mb-2"
      data-testid="hazardCard"
    >
      <header
        class="flex flex-1 text-brand-gray-80 py-2 pr-2 mr-1 font-semibold"
      >
        hazard 123
      </header>
      <div
        class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded mt-2 first:mt-0"
        data-testid="controlCard"
      >
        <span
          class="flex-1"
        >
          control name
        </span>
      </div>
      <div
        class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded mt-2 first:mt-0"
        data-testid="controlCard"
      >
        <span
          class="flex-1"
        >
          control name 2
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
